spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/bootstrap/KafkaBinderMeterRegistryTest.java

```diff
@@ -0,0 +1,105 @@
+
+import io.micrometer.core.instrument.MeterRegistry;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.test.rule.EmbeddedKafkaRule;
+
+import java.util.function.Function;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+/**
+ * @author Soby Chacko
+ */
+public class KafkaBinderMeterRegistryTest {
+
+    @ClassRule
+    public static EmbeddedKafkaRule embeddedKafka = new EmbeddedKafkaRule(1, true, 10);
+
+    @Test
+    public void testMetricsWorkWithSingleBinder() {
+        ConfigurableApplicationContext applicationContext = new SpringApplicationBuilder(SimpleApplication.class)
+                .web(WebApplicationType.NONE)
+                .run("--spring.cloud.stream.bindings.uppercase-in-0.destination=inputTopic",
+                        "--spring.cloud.stream.bindings.uppercase-in-0.group=inputGroup",
+                        "--spring.cloud.stream.bindings.uppercase-in-0.binder=kafka1",
+                        "--spring.cloud.stream.bindings.uppercase-output-0.destination=outputTopic",
+                        "--spring.cloud.stream.bindings.uppercase-output-0.binder=kafka1",
+                        "--spring.cloud.stream.binders.kafka1.type=kafka");
+
+        final MeterRegistry meterRegistry = applicationContext.getBean(MeterRegistry.class);
+
+        assertMeterRegistry(meterRegistry);
+
+        applicationContext.close();
+    }
+
+    @Test
+    public void testMetricsWorkWithMultiBinders() {
+        ConfigurableApplicationContext applicationContext = new SpringApplicationBuilder(SimpleApplication.class)
+                .web(WebApplicationType.NONE)
+                .run("--spring.cloud.stream.bindings.uppercase-in-0.destination=inputTopic",
+                        "--spring.cloud.stream.bindings.uppercase-in-0.group=inputGroup",
+                        "--spring.cloud.stream.bindings.uppercase-in-0.binder=kafka1",
+                        "--spring.cloud.stream.bindings.uppercase-output-0.destination=outputTopic",
+                        "--spring.cloud.stream.bindings.uppercase-output-0.binder=kafka2",
+                        "--spring.cloud.stream.binders.kafka1.type=kafka",
+                        "--spring.cloud.stream.binders.kafka2.type=kafka",
+                        "--spring.cloud.stream.default.binder=kafka1");
+
+        final MeterRegistry meterRegistry = applicationContext.getBean(MeterRegistry.class);
+
+        assertMeterRegistry(meterRegistry);
+
+        applicationContext.close();
+    }
+
+    private void assertMeterRegistry(MeterRegistry meterRegistry) {
```

Comment:
Given a valid meter registry we want to assert on
1) Kafka binder metrics
2) Producer Metrics
3) Consumer Metrics